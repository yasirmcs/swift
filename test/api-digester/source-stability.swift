// REQUIRES: OS=macosx
// RUN: rm -rf %S/tmp && mkdir %S/tmp && mkdir %S/tmp/module-cache && mkdir %S/tmp/dummy.sdk
// RUN: %api-digester -dump-sdk -module Swift -o %S/tmp/current-stdlib.json -module-cache-path %S/tmp/module-cache -sdk %S/tmp/dummy.sdk -swift-version 3
// RUN: %api-digester -diagnose-sdk -input-paths %S/stdlib-stable.json -input-paths %S/tmp/current-stdlib.json >> %S/tmp/changes.txt
// RUN: %clang -E -P -x c %S/source-stability.swift.expected -o - | sed '/^\s*$/d' > %S/tmp/source-stability.swift.expected
// RUN: %clang -E -P -x c %S/tmp/changes.txt -o - | sed '/^\s*$/d' > %S/tmp/changes.txt.tmp
// RUN: diff -u %S/tmp/source-stability.swift.expected %S/tmp/changes.txt.tmp
// RUN: rm -rf %S/tmp
