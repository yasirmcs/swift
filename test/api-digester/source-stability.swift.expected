/* DictionaryIndex => Dictionary.Index,
 * SetIndex => Set.Index
 *
 * Note: The old names are still available via generic typealiases,
 * however swift-api-digester doesn't pick that up yet. */
TypeAlias Dictionary.Index has been removed
TypeAlias Set.Index has been removed
Struct DictionaryIndex has been removed
Struct SetIndex has been removed

/* Removed declarations */
Protocol BidirectionalIndexable has been removed (deprecated)
Protocol ExpressibleByStringInterpolation has been removed (deprecated)
Protocol Indexable has been removed (deprecated)
Protocol IndexableBase has been removed (deprecated)
Protocol MutableIndexable has been removed (deprecated)
Protocol RandomAccessIndexable has been removed (deprecated)
Protocol RangeReplaceableIndexable has been removed (deprecated)

/* More DictionaryIndex / SetIndex */
Var Dictionary.endIndex has declared type change from DictionaryIndex<Key, Value> to Dictionary<Key, Value>.Index
Var Dictionary.startIndex has declared type change from DictionaryIndex<Key, Value> to Dictionary<Key, Value>.Index
Var Set.endIndex has declared type change from SetIndex<Element> to Set<Element>.Index
Var Set.startIndex has declared type change from SetIndex<Element> to Set<Element>.Index
Func Dictionary.index(after:) has return type change from DictionaryIndex<Key, Value> to Dictionary<Key, Value>.Index
Func Dictionary.index(forKey:) has return type change from DictionaryIndex<Key, Value>? to Dictionary<Key, Value>.Index?
Func Dictionary.remove(at:) has 1st parameter type change from DictionaryIndex<Key, Value> to Dictionary<Key, Value>.Index

/* False positives due to archetype -> interface type change */
Func OptionSet.insert(_:) has return type change from (inserted: Bool, memberAfterInsert: Self) to (inserted: Bool, memberAfterInsert: Self.Element)
Func OptionSet.remove(_:) has return type change from Self? to Self.Element?
Func OptionSet.update(with:) has return type change from Self? to Self.Element?
Func RandomAccessCollection.distance(from:to:) has return type change from Self.IndexDistance to Self.Index.Stride

/* More DictionaryIndex / SetIndex */
Func Set.index(after:) has return type change from SetIndex<Element> to Set<Element>.Index
Func Set.index(of:) has return type change from SetIndex<Element>? to Set<Element>.Index?
Func Set.remove(at:) has 1st parameter type change from SetIndex<Element> to Set<Element>.Index

/* Function type change */
Func UnsafePointer.withMemoryRebound(to:capacity:_:) has 3rd parameter type change from (UnsafeMutablePointer<T>) throws -> Result to (UnsafePointer<T>) throws -> Result

